### PR TITLE
influx CLI 2.4 release notes

### LINKED ISSUE #4170: Influx CLI 2.4 Release Notes
### Features
- Allow setting membership type in with `influx org members add` where only `member` was available before
- Add back the InfluxQL Shell (REPL) from v1 with an enhanced experience
- **[Cloud only]** Allow for calling [Invokable Scripts](https://docs.influxdata.com/influxdb/cloud/api-guide/api-invokable-scripts/)
- **[OSS only]** Add username and password authentication as alternative to token
- **[OSS only]** Add username and password signin
- Add `--extra-http-header` flag
- Add `skipRowOnError` handling for raw line protocol files
- Prevent deletion of replications when deleting remotes. Support this with a `--delete-replications` flag to the `influx remote delete` command instead of automatically deleting replications associated with that remote.

### Maintence
- Upgrade to Go 1.18.3

### Bug fixes
- User and org permissions no longer scoped to an individual org. Removes an unwanted restriction
- Properly handle tokens starting with a hyphen (`-`) passed to CLI without `=` (i.e., `influx cmd --token <token>`)
- Add mutual exclusion for `--org`/`$INFLUX_ORG` and `org_id`/`$INFLUX_ORG_ID`. Eliminate confusion on which to use and whether env vars were overriding flags
- Show remotes and replications flags in `auth create` command

### PR DESCRIPTION
**Pull request by sanderson**
Closes #4170 

- [x] Rebased/mergeable

### CODE REVIEW

File: content/influxdb/v2.4/reference/release-notes/influx-cli.md [hunk 1]
```diff
@@ -8,112 +8,142 @@ menu:
...
 
-## v2.3 [2022-04-08]
+## v2.4.0
 
-## Features
+### Features
+
+- Set membership type to member or owner with
+  [`influx org members add`](/influxdb/v2.4/reference/cli/influx/org/members/add/).
+- Add the [InfluxQL Shell (REPL)](/influxdb/v2.4/reference/cli/influx/v1/shell/).
```
Comment (kelseiv) [inline content/influxdb/v2.4/reference/release-notes/influx-cli.md:17]: looks like broken link, maybe just cause that new file is in another PR?

**Review by kelseiv — APPROVED** [review]

File: content/influxdb/v2.4/reference/release-notes/influx-cli.md [hunk 2]
```diff
@@ -8,112 +8,142 @@ menu:
     name: influx CLI 
 ---
 
-## v2.3 [2022-04-08]
+## v2.4.0
```
Comment (kelseiv) [inline content/influxdb/v2.4/reference/release-notes/influx-cli.md:11]: release date TBD

Comment (sanderson) [inline content/influxdb/v2.4/reference/release-notes/influx-cli.md:11]:
```suggestion
## v2.4.0 [TBD]
```

File: content/influxdb/v2.4/reference/release-notes/influx-cli.md (hunk 1 above)
Comment (sanderson) [reply content/influxdb/v2.4/reference/release-notes/influx-cli.md:17]: Yeah, this will be fixed with #4263.

**Pull request merged**